documentation/src/main/asciidoc/user_guide/server_protocols_hotrod.adoc

```diff
@@ -917,6 +917,38 @@ In order to avoid creating a TCP connection on each request (which is a costly o
 
 The number of connections per server, total number of connections, how long should a connection be kept idle in the pool before being closed - all these (and more) can be configured. Please refer to the javadoc of link:{javadocroot}/org/infinispan/client/hotrod/RemoteCacheManager.html[RemoteCacheManager] for a list of all possible configuration elements.
 
+==== Using internal/external IP addresses
+
+In some cases (e.g. OpenStack or Kubernetes with services exposed using load balancers) servers use different IP addresses for clustering and different for communicating with the client.
```

Comment:
Hmmmm, I have doubts that unless we provide some implementations that can easily be tailored to the environments, people will actually use this.

As a user, I can get around this issue by having some kind of HTTP entry point in the container environment and from that access the different Hot Rod servers without the need of any of these issues.

My point is: if you want people to use this, they should need to do as little as possible to get it working with the most popular container envs. If I were you, at the very least, I'd try to provide something that can easily be used from Kubernetes.